src/client/ducks/repositories/selectors/index.js

```diff
@@ -72,3 +72,13 @@ export const getRepositoryId = createSelector(
     _.findKey(entities, (v) => (v.fullName === repoName))
 )
 
+export const getChangelogFetchStatus = statusFetchFactory(types.FETCH_CHANGELOG)
+export const getChangelog = (state: Object, props: Object) => {
+  const changesets = _.get(state, ['entities', 'changesets'], [])[undefined]
```

Comment:
I'll fix normalization for you, but this should be composed selector, move out `changesets` to separate selector. 

And I misunderstood your question about sorting before, definitely should be sorted by commit date, descending.